packages/server/src/__tests__/ApolloServer.test.ts

```diff
@@ -567,3 +573,51 @@ describe('ApolloServer addPlugin', () => {
     await server.stop();
   });
 });
+
+it('TypedQueryDocumentNode', async () => {
+  const server = new ApolloServer({
+    typeDefs: `type Query {
+    foo(arg: Int!): String
+  }`,
+    resolvers: {
+      Query: {
+        foo(_, { arg }) {
+          return `yay ${arg}`;
+        },
+      },
+    },
+  });
+  await server.start();
+
+  const query = parse(
+    'query Q($a: Int!) {foo(arg: $a)}',
+  ) as TypedQueryDocumentNode<{ foo: string | null }, { a: number }>;
+
+  const response = await server.executeOperation({
+    query,
+    variables: { a: 1 },
+  });
+  assert(response.body.kind === 'single');
+  expect(response.body.singleResult.data?.foo).toBe('yay 1');
+  // bar is not part of the data type.
+  // @ts-expect-error
+  expect(response.body.singleResult.data?.bar).toBe(undefined);
+
+  if (1 + 1 === 3) {
```

Comment:
lol works for me

Reply:
`if (false)` bad for lint?

Reply:
tsc finds dead code and complains

Reply:
oh you're right, it's lint

Reply:
no, it's tsc but only in vscode not in CLI?

Reply:
ah, the default value for `allowUnreachableCode` is:

> undefined (default) provide suggestions as warnings to editors